File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDCity.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOCity 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  city_id ,city ,country_id ,last_update  from city";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClCity city)  
        {
            String stSql = "Select  city_id ,city ,country_id ,last_update  from city";
            stSql += " Where  city_id='" + city.CityId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        city.City = dt.Rows[0]["city"].ToString();
        city.CountryId = new ClCountry(int.Parse(dt.Rows[0]["country_id"].ToString()));
        city.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClCity city)  
        {
            String stSql = "insert into City (" 
                       + " city_id ,city ,country_id ,last_update" 
                           + " )values (" 
            + "'" + city.CityId + "'" 
                     + ",'" + city.City + "'" 
                     + ",'" + city.CountryId.CountryId + "'" 
                     + ",'" + city.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClCity city) {
            
            String stSql  =  "delete from City " 
             + " Where  city_id='" + city.CityId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClCity city ) {
            String stSql  =  "update City SET " 
             + " city = '" + city.City + "'" 
                             + ",country_id = '" + city.CountryId.CountryId + "'" 
                             + ",last_update = '" + city.LastUpdate + "'" 
             + " Where  city_id='" + city.CityId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}

File: Sakila_Net/Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDCity.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOCity
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from city"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( city as ClCity) as Boolean
             Dim StSql as String = "Select * from city"
            StSql &= " Where  city_id='" & city.dataCityId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        city.DataCity = dt.Rows(0)("city").ToString()
        city.DataCountryId = new ClCountry(Integer.Parse(dt.Rows(0)("country_id")))
        city.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( city as ClCity)  as Integer
            Dim stSql as String = "insert into City (" _
                       & " city_id ,city ,country_id ,last_update" _
                           & " )values (" _
            & "'" & city.dataCityId & "'" _
                     & ",'" & city.dataCity & "'" _
                     & ",'" & city.dataCountryId.datacountryId & "'" _
                     & ",'" & city.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( city as ClCity) as Integer
            
            Dim stSql as String =  "delete from City " _
             & " Where  city_id='" & city.dataCityId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(city as ClCity) as Integer
            Dim stSql as String =  "update City SET " _
             & " city = '" & city.DataCity & "'" _
                             & ",country_id = '" & city.DataCountryId.DataCountryId & "'" _
                             & ",last_update = '" & city.DataLastUpdate & "'" _
             & " Where  city_id='" & city.dataCityId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  
